### DISCUSSION TITLE
How to handle Time fields?

### DISCUSSION THREAD

**Discussion by ludo237** [pocketbase/pocketbase#1143] [Q&A] [answered]
Hi everyone, I'm trying out Pocketbase and I got stuck on an example collection for `shifts`. The idea is to have a select field called `day` and then two **time only** fields called `from` and `to`. The ideal format should be `H:i` but this is not possible with the Datetime field provided. 
What is the best approach in this case? Should I go for a simple text field and regex? What type of regex do you suggest?

**Comment by dschissler** [comment]
This seems like an issue of client side formatting is all.  If you don't need to store some minutes or seconds then perhaps you could just set them each to "00"?  Otherwise if it's not really a time, but is in fact a duration then perhaps you should store that as a number.  Then format somehow via code.

**Comment by ganigeorgiev** [comment, marked as answer]
@ludo237 There are several ways you can handle this. You can use your approach with the `text` field for `from` and `to` and use a regex like:

```regex
// 24hour format
^(?:2[0-3]|[01][0-9]):[0-5][0-9]
```
(_I haven't tested it; I've just copied the first google result from https://stackoverflow.com/questions/3964972/validate-that-input-is-in-this-time-format-hhmm_).

But I would recommend another approach - instead of 3 fields as in your example (`day`, `from`, `to`) to have only 2 fields - `from` and `to` that will contain a full datetime for each period (this allows also handling "shifts" that span across multiple days). On the frontend you can still have 3 different fields, but when submitting your data to PocketBase you'll have to "join" them in the background.

**Reply by ludo237** [reply]
The regex works fine, thanks. Also I like your idea and approach suggested I'll take that into consideration. Keep in mind I'm just tinkering with Pocketbase in order to see if it makes sense for an actual project
